### DISCUSSION TITLE
add_edge takes 3 positional arguments but 4 were given

### DISCUSSION THREAD

**Discussion by jqdoumen** [networkx/networkx#7470] [General]
Want to simply add a weight to an edge ...

```
G = nx.Graph()
for record in results:
    source = record["source"]
    target = record["target"]
    weight = record["weight"]
    G.add_edge(source, target, weight)
```

and get as error

`TypeError: add_edge() takes 3 positional arguments but 4 were given`

Data looks all right though ...

```
G = nx.Graph()
for record in results:
    print(record)

'25'	'1'	6
'26'	'1'	6
'2'	'1'	6
'24'	'1'	6
'28'	'1'	2
```

Seen a few occurrences of this error, but date back from several years ago, and nothing that seems to solve the issue.

Any ideas?

**Comment by eriknw** [comment]
Looking at the function signature of `Graph.add_edge`
https://github.com/networkx/networkx/blob/2a03462ea94b1ca70b8ab9cb5e04b85b2a146833/networkx/classes/graph.py#L903
we see that the three expected arguments are `self`, `u_of_edge` and `v_of_edge`.

Notably, any edge attribute (often called the weight) must be given as keyword arguments, not as positional arguments. So, your example should probably be something like:
```python
G = nx.Graph()
for record in results:
    source = record["source"]
    target = record["target"]
    weight = record["weight"]
    G.add_edge(source, target, weight=weight)
```
Multigraphs may accept another positional argument for the edge key as seen here
https://github.com/networkx/networkx/blob/2a03462ea94b1ca70b8ab9cb5e04b85b2a146833/networkx/classes/multigraph.py#L430

**Comment by jqdoumen** [comment]
Thx. Proposal works !
